src/components/RepositoryNavigation.js

```diff
@@ -0,0 +1,177 @@
+    const repo = this.state.repository.queryStringMap.repo;
+    const path = newQueryStringMap.path;
+
+    if (newQueryStringMap !== this.state.repository.queryStringMap) {
+      const json = await getContents(owner, repo, path);
+      this.setState({
+        repository: {
+          isLoaded: true,
+          queryStringMap: newQueryStringMap,
+          contents: json
+        }
+      });
+    }
+  }
+
+  render() {
+    const { isLoaded } = this.state.repository;
+    if (!isLoaded) {
+      return (
+        <div className="Markdown">Loading...</div>
+      );
+    } else {
+      const listGroupItems = [];
+      const contents = this.state.repository.contents;
+      const queryStringMap = this.state.repository.queryStringMap;
+
+      contents.forEach((item) => {
+        // Create a new copy of the old query string map, then modify it with references to each
+        // item's relative path
+        const resourceQueryStringCopy = { ...queryStringMap };
+        const key = `list-group-item-${item.name}`;
+        resourceQueryStringCopy.path = `/${item.path}`;
+        const newQueryString = queryString.stringify(resourceQueryStringCopy);
+        listGroupItems.push(
+          item.type === 'dir' ? (
+            <LinkContainer to={`${REPO_RESOURCE}?${newQueryString}`} key={key}>
+              <ListGroupItem action>
+                <i>{`/${item.name}`}</i>
+              </ListGroupItem>
+            </LinkContainer>
+          ) : item.type === 'file' ? (
+            <LinkContainer to={`${RENDER_RESOURCE}?${newQueryString}`} key={key}>
+              <ListGroupItem action>
+                {item.name}
+              </ListGroupItem>
+            </LinkContainer>
+          ) : <div>File type {item.type} not supported</div>
+        );
+      });
+      const breadcrumbsComponent = getBreadcrumbsComponent(this.state.repository.queryStringMap);
+
+      return (
+        <>
+          <h2>Repository Contents</h2>
+            {breadcrumbsComponent}
+          <ListGroup>
+            {listGroupItems}
+          </ListGroup>
+        </>
+      );
+    }
+  }
+}
+
+/**
+ * Returns a list of breadcrumbs based on current path of navigation
+ * @param {Object} queryStringMap JSON map of query string
+ * @return {Array} List of BreadcrumbItems
+ */
+function getBreadcrumbsComponent(queryStringMap) {
+  const breadcrumbItems = [];
+  const baseLink = `${REPO_RESOURCE}`;
+  const keyBase = 'breadcrumb-item-';
+
+  const subDirectoriesArr = queryStringMap.path.split('/');
+  // discord root case, as '/' gets split into ['', '']
+  if (subDirectoriesArr[1] === '') {
+    subDirectoriesArr.shift();
+  }
+  // using currDir to rebuild intermediary paths for each subdirectory in queryStringMap
+  const currDir = '';
+  for (let i = 0; i < subDirectoriesArr.length; i++) {
+    const directory = subDirectoriesArr[i];
+    const queryStringMapCopy = { ...queryStringMap };
+    queryStringMapCopy.path = currDir.concat('/', directory);
```

Comment:
`currDir` isn't necessary here. This can just be ``queryStringMapCopy.path = `/${directory}`;``. 

If you want to be building a path, you need to set `currDir`. `.concat()` is not in-place, it returns the result.

```javascript
let currDir = '';
currDir = currDir.concat('/', directory);
queryStringMapCopy.path = currDir;
```